test/e2e/fixtures/openapi3-provider-with-keywords.yaml

```diff
@@ -0,0 +1,72 @@
+openapi: 3.1.0
+info:
+  title: anyOf / oneOf / allOf example
+  description: Demonstrate the use of these logical keywordsa
+  version: 1.0.0
+paths:
+  /pet:
+    get:
+      summary: Gets a random pet
+      description: Returns a random pet
+      operationId: getRandomPet
+      responses:
+        "200":
+          description: A random pet
+          content:
+            "application/json":
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/Cat'
+                  - $ref: '#/components/schemas/Dog'
+                discriminator:
+                  propertyName: pet_type
+                required: # this is NOT removed because oneOf needs it
+                  - pet_type
+components:
+  schemas:
+    Pet:
+      type: object
+      required: # this is ignored, because provider returns superset of what consumer uses
```

Comment:
This `required` is transformed away as per usual. But the `required` next to `oneOf` is retained.